### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by persiaAziz**
This is a simple version of the pipeline code. This will grow more in the later iterations. Right now enabling full deploy is not secure

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
```
Comment (tniek) [inline azure-pipelines.yml]: Maybe need to re-write this paragraph, since it's a public repo.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    vmImage: "ubuntu-16.04"
+  displayName: "Test deployment of single node hana"
+  steps:
+  - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
+    clean: "all"
+    path: "sap-hana"  # path to check out source code, relative to the agent's build directory
+    persistCredentials: boolean  # set to 'true' to leave the OAuth token in the Git config after the initial fetch
+  - script: |
+      ssh-keygen -b 4096 -t rsa -f /tmp/sshkey -q -N ""
+      cp deploy/vm/test/terraform.tfvars deploy/vm/modules/single_node_hana
```
Comment (tniek) [inline azure-pipelines.yml]: This is ok for the current version of our templates, but this structure will change (i.e. one single TF module as entry point, and one JSON config for both TF and Ansible). Would it make sense to abstract this and make it more general?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      TF_VAR_pw_os_sapadm: $(hana-int-pw-sap-adm)
+      TF_VAR_pw_hacluster: $(hana-int-pw-sap-adm)
+      AZURE_CLIENT_ID: $(hana-pipeline-spn-id)
+      AZURE_SECRET: $(hana-pipeline-spn-pw)
+      AZURE_TENANT: $(landscape-tenant)
+      AZURE_SUBSCRIPTION_ID: $(landscape-subscription)
+      ARM_CLIENT_ID: $(hana-pipeline-spn-id)
+      ARM_CLIENT_SECRET: $(hana-pipeline-spn-pw)
+      ARM_TENANT_ID: $(landscape-tenant)
+      ARM_SUBSCRIPTION_ID: $(landscape-subscription)
```
Comment (tniek) [inline azure-pipelines.yml]: nit: newline at EOF

File: deploy/vm/test/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,11 @@
+az_region = "westus"
+vm_user = "azureuser"
+az_domain_name = "singlehana"
```
Comment (tniek) [inline deploy/vm/test/terraform.tfvars]: Probably need to go for something more "unique" to avoid deployment failures

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# https://aka.ms/yaml
+
+jobs:
+- job: "single_node_hana"
+  variables:
+    - group: azure-config-variables
+    - group: azure-sap-hana-pipeline-secrets
+  pool:
+    vmImage: "ubuntu-16.04"
+  displayName: "Test deployment of single node hana"
```
Comment (tniek) [inline azure-pipelines.yml]: nit: "single-node HANA"

File: azure-pipelines.yml (hunk 1 above)
Comment (persiaAziz) [reply azure-pipelines.yml]: Yah this is the default headlines that azure pipelines put. I forgot to change that.

File: azure-pipelines.yml (hunk 2 above)
Comment (persiaAziz) [reply azure-pipelines.yml]: Our current version is dependent on tfvars. I will need the exact format and schema for the new structure. I think we should keep it like this for our version. We can change accordingly when the newer version arrives.

File: deploy/vm/test/terraform.tfvars (hunk 4 above)
Comment (persiaAziz) [reply deploy/vm/test/terraform.tfvars]: good catch

**Pull request merged**